DegreeProject/Assets/Scripts/Enemy/States/PatrolState.cs

```diff
@@ -80,58 +79,33 @@ private void HorizontalCollision(AI owner)
     {
         Vector2 originRight = new Vector2(owner.transform.position.x + 0.5f, owner.transform.position.y + 0.65f);
         Vector2 originLeft = new Vector2(owner.transform.position.x - 0.5f, owner.transform.position.y + 0.65f);
-        float distance = 0.15f;
+        Vector2 origin = direction.x == 1 ? originRight : originLeft;
```

Comment:
Att köra equals to med floats är väldigt risky, då `1.00000001 != 1`. Nu sätter du distance till Vector2.right alt. .left, men det är bra att alltid jobba med lite leeway när du jämför floats. I detta exempel kan du t.ex. kolla om `direction.x > 0f`